--- ./patchsource/www/rega/esp/datapointconfigurator.fn.orig
+++ ./patchsource/www/rega/esp/datapointconfigurator.fn
@@ -99,6 +99,11 @@
     var CN_BRIGHTNESS_TRANSMITTER = "BRIGHTNESS_TRANSMITTER.";
     var CN_BTN_SHORT_ONLY  = "BTN_SHORT_ONLY.";
     var CN_BUTTON          = "BUTTON.";
+    var CN_RFID_BUTTON     = "RFIDBTN.";
+    var CN_IBTN_BUTTON     = "IBTN.";
+    var CN_SERVO           = "SERVO.";
+    var CN_PFS             = "PFS.";
+    var CN_LDRSC           = "LDRSC.";
     var CN_CAPACITIVE_FILLING_SENSOR  = "CAPACITIVE_FILLING_LEVEL_SENSOR.";
     var CN_CLIMATE_CONTROL_FLOOR_PUMP_TRANSCEIVER = "CLIMATECONTROL_FLOOR_PUMP_TRANSCEIVER.";
     var CN_CLIMATE_CONTROL_FLOOR_TRANSCEIVER = "CLIMATECONTROL_FLOOR_TRANSCEIVER.";
@@ -369,7 +374,58 @@
             sLastControlName = CN_BTN_SHORT_ONLY;
             Call("/esp/controls/button.fn::CreateKeyShortOnly()");
           }
+          
+          !# RFID
+          bIsControl = ( sControlName.Find(CN_RFID_BUTTON) > -1 );
+          if( bIsControl && (sLastControlName!=CN_RFID_BUTTON) )
+          {
+            WriteLine( "<script>conInfo('Control CN_RFID_BUTTON found.');</script>" );
+            isKnownControl = true;
+            sLastControlName = CN_RFID_BUTTON;
+            Call("/esp/controls/rfid_rc.fn::CreateRfidKey()");
+          }
 
+          !# iBUTTON
+          bIsControl = ( sControlName.Find(CN_IBTN_BUTTON) > -1 );
+          if( bIsControl && (sLastControlName!=CN_IBTN_BUTTON) )
+          {
+            WriteLine( "<script>conInfo('Control CN_IBTN_BUTTON found.');</script>" );
+            isKnownControl = true;
+            sLastControlName = CN_IBTN_BUTTON;
+            Call("/esp/controls/ibutton.fn::CreateIButtonKey()");
+          }
+                    
+          !# SERVO
+          bIsControl = ( sControlName.Find(CN_SERVO) > -1 );
+          if( bIsControl && (sLastControlName!=CN_SERVO) )
+          {
+            object ch = dom.GetObject(chnId);
+            isKnownControl = true;
+            sLastControlName = CN_SERVO;
+            WriteLine( "<script>conInfo('Control SERVO found.');</script>" );
+            Call("/esp/controls/servo.fn::CreateServoActuator()");
+          } 
+          
+          !# PFS (Pet Flap Sensor - Tierklappensensor)
+          bIsControl = ( sControlName.Find(CN_PFS) > -1 );
+          if( bIsControl && (sLastControlName!=CN_PFS) )
+          {
+            WriteLine( "<script>conInfo('Control PFS found.');</script>" );
+            isKnownControl = true;
+            sLastControlName = CN_PFS;
+            Call("/esp/controls/pfs.fn::CreatePFSensor()");
+          }
+          
+          !# LDRSC
+          bIsControl = ( sControlName.Find(CN_LDRSC) > -1 );
+          if( bIsControl && (sLastControlName!=CN_LDRSC) )
+          {
+            WriteLine( "<script>conInfo('Control LDRSC found.');</script>" );
+            isKnownControl = true;
+            sLastControlName = CN_LDRSC;
+            Call("/esp/controls/ldrsc.fn::CreateLDRSC()");
+          }
+          
           !# BUTTON
           bIsControl = ( sControlName.Find(CN_BUTTON) > -1 );
           if( bIsControl && (sLastControlName!=CN_BUTTON) )
