--- ./patchsource/www/rega/esp/datapointconfigurator.fn.orig
+++ ./patchsource/www/rega/esp/datapointconfigurator.fn
@@ -99,6 +99,16 @@
     var CN_BRIGHTNESS_TRANSMITTER = "BRIGHTNESS_TRANSMITTER.";
     var CN_BTN_SHORT_ONLY  = "BTN_SHORT_ONLY.";
     var CN_BUTTON          = "BUTTON.";
+    var CN_RFID_BUTTON     = "RFIDBTN.";
+    var CN_IRRC_BUTTON     = "IRRCBTN.";
+    var CN_IBTN_BUTTON     = "IBTN.";
+    var CN_SERVO           = "SERVO.";
+    var CN_FAN             = "FAN.";
+    var CN_PFS             = "PFS.";
+    var CN_LDRSC           = "LDRSC.";
+    var CN_CRGSC           = "CRGSC.";
+    var CN_PWRSC           = "PWRSC.";
+    var CN_CONDSC          = "CONDSC.";
     var CN_CAPACITIVE_FILLING_SENSOR  = "CAPACITIVE_FILLING_LEVEL_SENSOR.";
     var CN_CLIMATE_CONTROL_FLOOR_PUMP_TRANSCEIVER = "CLIMATECONTROL_FLOOR_PUMP_TRANSCEIVER.";
     var CN_CLIMATE_CONTROL_FLOOR_TRANSCEIVER = "CLIMATECONTROL_FLOOR_TRANSCEIVER.";
@@ -369,7 +379,109 @@
             sLastControlName = CN_BTN_SHORT_ONLY;
             Call("/esp/controls/button.fn::CreateKeyShortOnly()");
           }
-
+          
+          !# RFID
+          bIsControl = ( sControlName.Find(CN_RFID_BUTTON) > -1 );
+          if( bIsControl && (sLastControlName!=CN_RFID_BUTTON) )
+          {
+            WriteLine( "<script>conInfo('Control CN_RFID_BUTTON found.');</script>" );
+            isKnownControl = true;
+            sLastControlName = CN_RFID_BUTTON;
+            Call("/esp/controls/rfid_rc.fn::CreateRfidKey()");
+          }
+                    
+          !# IRRC
+          bIsControl = ( sControlName.Find(CN_IRRC_BUTTON) > -1 );
+          if( bIsControl && (sLastControlName!=CN_IRRC_BUTTON) )
+          {
+            WriteLine( "<script>conInfo('Control CN_IRRC_BUTTON found.');</script>" );
+            isKnownControl = true;
+            sLastControlName = CN_IRRC_BUTTON;
+            Call("/esp/controls/ir_rc.fn::CreateIRKey()");
+          }
+          
+          !# iBUTTON
+          bIsControl = ( sControlName.Find(CN_IBTN_BUTTON) > -1 );
+          if( bIsControl && (sLastControlName!=CN_IBTN_BUTTON) )
+          {
+            WriteLine( "<script>conInfo('Control CN_IBTN_BUTTON found.');</script>" );
+            isKnownControl = true;
+            sLastControlName = CN_IBTN_BUTTON;
+            Call("/esp/controls/ibutton.fn::CreateIButtonKey()");
+          }
+                    
+          !# SERVO
+          bIsControl = ( sControlName.Find(CN_SERVO) > -1 );
+          if( bIsControl && (sLastControlName!=CN_SERVO) )
+          {
+            object ch = dom.GetObject(chnId);
+            isKnownControl = true;
+            sLastControlName = CN_SERVO;
+            WriteLine( "<script>conInfo('Control SERVO found.');</script>" );
+            Call("/esp/controls/servo.fn::CreateServoActuator()");
+          } 
+          
+          !# FAN
+          bIsControl = ( sControlName.Find(CN_FAN) > -1 );
+          if( bIsControl && (sLastControlName!=CN_FAN) )
+          {
+            object ch = dom.GetObject(chnId);
+            isKnownControl = true;
+            sLastControlName = CN_FAN;
+            WriteLine( "<script>conInfo('Control FAN found.');</script>" );
+            Call("/esp/controls/fan.fn::CreateFanActuator()");
+          } 
+                    
+          !# PFS (Pet Flap Sensor - Tierklappensensor)
+          bIsControl = ( sControlName.Find(CN_PFS) > -1 );
+          if( bIsControl && (sLastControlName!=CN_PFS) )
+          {
+            WriteLine( "<script>conInfo('Control PFS found.');</script>" );
+            isKnownControl = true;
+            sLastControlName = CN_PFS;
+            Call("/esp/controls/pfs.fn::CreatePFSensor()");
+          }
+          
+          !# LDRSC
+          bIsControl = ( sControlName.Find(CN_LDRSC) > -1 );
+          if( bIsControl && (sLastControlName!=CN_LDRSC) )
+          {
+            WriteLine( "<script>conInfo('Control LDRSC found.');</script>" );
+            isKnownControl = true;
+            sLastControlName = CN_LDRSC;
+            Call("/esp/controls/ldrsc.fn::CreateLDRSC()");
+          }
+          
+          !# CRGSC
+          bIsControl = ( sControlName.Find(CN_CRGSC) > -1 );
+          if( bIsControl && (sLastControlName!=CN_CRGSC) )
+          {
+            WriteLine( "<script>conInfo('Control CRGSC found.');</script>" );
+            isKnownControl = true;
+            sLastControlName = CN_CRGSC;
+            Call("/esp/controls/crgsc.fn::CreateCRGSC()");
+          }
+          
+          !# PWRSC
+          bIsControl = ( sControlName.Find(CN_PWRSC) > -1 );
+          if( bIsControl && (sLastControlName!=CN_PWRSC) )
+          {
+            WriteLine( "<script>conInfo('Control PWRSC found.');</script>" );
+            isKnownControl = true;
+            sLastControlName = CN_PWRSC;
+            Call("/esp/controls/pwrsc.fn::CreatePWRSC()");
+          }
+          
+          !# CONDSC
+          bIsControl = ( sControlName.Find(CN_CONDSC) > -1 );
+          if( bIsControl && (sLastControlName!=CN_CONDSC) )
+          {
+            WriteLine( "<script>conInfo('Control CONDSC found.');</script>" );
+            isKnownControl = true;
+            sLastControlName = CN_CONDSC;
+            Call("/esp/controls/condsc.fn::CreateCONDSC()");
+          }
+          
           !# BUTTON
           bIsControl = ( sControlName.Find(CN_BUTTON) > -1 );
           if( bIsControl && (sLastControlName!=CN_BUTTON) )
