--- ./patchsource/www/config/ic_deviceparameters.cgi.orig
+++ ./patchsource/www/config/ic_deviceparameters.cgi
@@ -891,6 +891,7 @@
         || ([string equal $ch_paramid "dw_controller_color_virt_ch_master"] == 1)
         || ([string equal $ch_paramid "dw_controller_brightness_ch_master"] == 1)
         || ([string equal $ch_paramid "dw_controller_brightness_virt_ch_master"] == 1)
+        || ([string equal $ch_paramid "hb_sw_tp_ch_master"] == 1)
         } {
          setInternalDeviceKey $ch_paramid
         }
@@ -909,7 +910,8 @@
         global receiver_address sender_address dev_descr_sender 
         set sender_address $ch_address
         set receiver_address $ch_address
-        incr counter
+        #incr counter
+        set counter [lindex [split $ch_descr(ADDRESS) ":"] 1]
         
         # internal key unlocked in xml-file?
         if {![catch {array set intKey [xmlrpc $iface_url($iface) getParamset [list string $dev_descr(ADDRESS)] [list string MASTER]]}]} {
