--- ./patchsource/www/webui/js/extern/sliderControl.js.orig
+++ ./patchsource/www/webui/js/extern/sliderControl.js
@@ -29,7 +29,7 @@
   if (factor) {this.factor = 100/(max-min);} else {this.factor = 1}
 
   this.n_minValue = 0;
-  this.n_maxValue = 100;
+  this.n_maxValue = (type == "Servo") ? 180 : 100;
   
   this.n_step = 0;
   this.s_form = null;
@@ -45,8 +45,8 @@
   this.n_pathLeft = 1;
   this.n_pathTop = 1;
   this.n_pathLength = 190;
-  this.s_imgControlGray = '/ise/img/slider/slider_grey.png';
-  this.s_imgControlGreen = '/ise/img/slider/slider_yellow.png';
+  this.s_imgControlGray = (type == "Servo") ?'/ise/img/slider/servoslider_grey.png':'/ise/img/slider/slider_grey.png';
+  this.s_imgControlGray = (type == "Servo") ?'/ise/img/slider/servoslider_grey.png':'/ise/img/slider/slider_yellow.png';
   this.s_imgSlider = '/ise/img/slider/slider.gif';  
   
   // register in the global collection  
