--- ./patchsource/www/rega/esp/datapointconfigurator.fn.orig	2018-10-31 21:12:18.000000000 +0100
+++ ./patchsource/www/rega/esp/datapointconfigurator.fn	2019-01-17 13:28:45.000000000 +0100
@@ -99,6 +99,8 @@
     var CN_BRIGHTNESS_TRANSMITTER = "BRIGHTNESS_TRANSMITTER.";
     var CN_BTN_SHORT_ONLY  = "BTN_SHORT_ONLY.";
     var CN_BUTTON          = "BUTTON.";
+    var CN_RFID_BUTTON     = "RFIDBTN.";
+    var CN_IBTN_BUTTON     = "IBTN.";
     var CN_CAPACITIVE_FILLING_SENSOR  = "CAPACITIVE_FILLING_LEVEL_SENSOR.";
     var CN_CLIMATE_CONTROL_FLOOR_PUMP_TRANSCEIVER = "CLIMATECONTROL_FLOOR_PUMP_TRANSCEIVER.";
     var CN_CLIMATE_CONTROL_FLOOR_TRANSCEIVER = "CLIMATECONTROL_FLOOR_TRANSCEIVER.";
@@ -364,7 +366,27 @@
             sLastControlName = CN_BTN_SHORT_ONLY;
             Call("/esp/controls/button.fn::CreateKeyShortOnly()");
           }
+          
+          !# RFID
+          bIsControl = ( sControlName.Find(CN_RFID_BUTTON) > -1 );
+          if( bIsControl && (sLastControlName!=CN_RFID_BUTTON) )
+          {
+            WriteLine( "<script>conInfo('Control CN_RFID_BUTTON found.');</script>" );
+            isKnownControl = true;
+            sLastControlName = CN_RFID_BUTTON;
+            Call("/esp/controls/rfid_rc.fn::CreateRfidKey()");
+          }
 
+          !# iBUTTON
+          bIsControl = ( sControlName.Find(CN_IBTN_BUTTON) > -1 );
+          if( bIsControl && (sLastControlName!=CN_IBTN_BUTTON) )
+          {
+            WriteLine( "<script>conInfo('Control CN_IBTN_BUTTON found.');</script>" );
+            isKnownControl = true;
+            sLastControlName = CN_IBTN_BUTTON;
+            Call("/esp/controls/ibutton.fn::CreateIButtonKey()");
+          }
+          
           !# BUTTON
           bIsControl = ( sControlName.Find(CN_BUTTON) > -1 );
           if( bIsControl && (sLastControlName!=CN_BUTTON) )
--- ./patchsource/www/webui/js/lang/de/translate.lang.js.orig	2018-10-26 07:01:22.000000000 +0200
+++ ./patchsource/www/webui/js/lang/de/translate.lang.js	2019-01-16 13:44:23.000000000 +0100
@@ -764,6 +764,8 @@
     "actionStatusControlLblOff" : "Aus",
     "actionStatusControlLblShortKeyPress" : "Kurzer<br/>Tastendruck",
     "actionStatusControlLblLongKeyPress" : "Langer<br/>Tastendruck",
+    "actionStatusControlLblShortRfidHold" : "kurzer<br/>Scan",
+    "actionStatusControlLblLongRfidHold" : "langer<br/>Scan",
     "commentActionStatusControlEnd" : "***********************************************",
     "commentPrgRules" : "***********************************************",
     "programsCreateProgram" : "Programmerstellung",
