--- ./patchsource/www/rega/esp/side.inc.orig	2018-12-05 21:12:40.000000000 +0100
+++ ./patchsource/www/rega/esp/side.inc	2019-02-13 10:43:57.000000000 +0100
@@ -12,9 +12,11 @@
   string hm_oucfm = "HM-OU-CFM-Pl";
   string hm_oucfmTW = "HM-OU-CFM-TW";
   string hm_oucmpcb = "HM-OU-CM-PCB";
+  string hb_oump3led = "HB-OU-MP3-LED";
   string hm_partyDialog = "HM-CC-RT-DN HM-TC-IT-WM-W-EU HM-CC-VG-1";
   string hm_statusDisplay = "HM-Dis-WM55";
   string hm_statusDisplayEPaper = "HM-Dis-EP-WM55";
+  string hb_statusDisplayEPaper = "HB-DIS-EP-42BW";
   string hm_rgbw_controller = "HM-LC-RGBW-WM";
   string hm_jalousieActor = "HM-LC-Ja1PBU";
   string heatingClimateControlTransceiver =  "HEATING_CLIMATECONTROL_TRANSCEIVER";
@@ -451,9 +453,11 @@
             integer oucfm = chLabel.Find(hm_oucfm);
             integer oucfmTW = chLabel.Find(hm_oucfmTW);
             integer oucmpcb = chLabel.Find(hm_oucmpcb);
+            integer oump3led = chLabel.Find(hb_oump3led);
             integer partyDialogDevice = hm_partyDialog.Find(chLabel);
             integer statusDisplayDevice = hm_statusDisplay.Find(chLabel);
             integer statusDisplayEPaperDevice = hm_statusDisplayEPaper.Find(chLabel);
+            integer statusHbDisplayEPaperDevice = hb_statusDisplayEPaper.Find(chLabel);
             integer rgbw_controller = chLabel.Find(hm_rgbw_controller);
             integer i_vir_lg_rgb_dim = chLabel.Find(s_vir_lg_rgb_dim);
             integer i_vir_lg_rgbw_dim = chLabel.Find(s_vir_lg_rgbw_dim);
@@ -462,6 +466,7 @@
 
             if ((statusDisplayDevice != -1)
               || (statusDisplayEPaperDevice != -1)
+              || (statusHbDisplayEPaperDevice != -1)
               || (rgbw_controller != -1)
               || ((jalousieActor != - 1) && (oDP.HSSID().Find("LEVEL_COMBINED") != -1))
               || (((i_vir_lg_rgb_dim != - 1) && (oDP.HSSID().Find("RGB") != -1)) || ((i_vir_lg_rgbw_dim != - 1) && (oDP.HSSID().Find("RGBW") != -1)) || (( i_vir_lg_group != - 1) && (oDP.HSSID().Find("RGBW") != -1)))
@@ -482,6 +487,13 @@
                 Write( '<input id="prgStringOptionValue'#oSD.ID()#'" onfocus="SetStatusDisplay('#oSD.ID()#',\'DIS-EP\');" type="text" class="SelectBox" size="10" value="'#iV#'" onchange="iseSingleDestination.SetValue('#oSD.ID()#',this.value,\'STRING\');" /> ' );
               }
 
+              if (statusHbDisplayEPaperDevice != -1) {
+                if(iV == "1") {
+                  ! This is the default string for 'all values not used'
+                  iV = "0x02,0x0A,0x0A,0x0A,0x0A,0x0A,0x0A,0x0A,0x0A,0x0A,0x0A,0x03";
+                }
+                Write( '<input id="prgStringOptionValue'#oSD.ID()#'" onfocus="SetStatusDisplay('#oSD.ID()#',\'DIS-EP42BW\');" type="text" class="SelectBox" size="10" value="'#iV#'" onchange="iseSingleDestination.SetValue('#oSD.ID()#',this.value,\'STRING\');" /> ' );
+              }
               if (rgbw_controller != -1) {
 
                 if (iV == "0") {
@@ -529,9 +541,11 @@
               && (oucfm == -1)
               && (oucfmTW == -1)
               && (oucmpcb == -1)
+              && (oump3led == -1)
               && (partyDialogDevice == -1)
               && (statusDisplayDevice == -1)
               && (statusDisplayEPaperDevice == -1)
+              && (statusHbDisplayEPaperDevice == -1)
               && (rgbw_controller == -1)
               && (jalousieActor == -1)
               && (i_vir_lg_rgb_dim == -1)
@@ -546,6 +560,9 @@
               if((oucfm != -1) || (oucfmTW != -1) || (oucmpcb != -1)) {
                 Write( '<span class="CLASS02401" onclick="ChangeOUCFMValue('#oSD.ID()#');"><img src="/ise/img/notepad.png" /></span>' );
               }
+              if(oump3led != -1) {
+                Write( '<span class="CLASS02401" onclick="ChangeHBOUCFMValue('#oSD.ID()#');"><img src="/ise/img/notepad.png" /></span>' );
+              }              
               if (partyDialogDevice != -1) {
                 Write( '<span id="setPartyMode" class="CLASS02401" onclick="SetPartyMode(this);"><img src="/ise/img/notepad.png" /></span>' );
               }
@@ -555,6 +572,9 @@
               if (statusDisplayEPaperDevice != -1) {
                 Write( '<span id="setStatusDisplayEPaper'#oSD.ID()#'" class="CLASS02401" onclick="SetStatusDisplay('#oSD.ID()#',\'DIS-EP\');"><img src="/ise/img/notepad.png" /></span>' );
               }
+              if (statusHbDisplayEPaperDevice != -1) {
+                Write( '<span id="setHbStatusDisplayEPaper'#oSD.ID()#'" class="CLASS02401" onclick="SetStatusDisplay('#oSD.ID()#',\'DIS-EP42BW\');"><img src="/ise/img/notepad.png" /></span>' );
+              }
               if (rgbw_controller != -1) {
                 Write( '<span id="setRGBWController'#oSD.ID()#'" class="CLASS02401" onclick="SetRGBWController('#oSD.ID()#',\''#oDP.HSSID()#'\');"><img src="/ise/img/notepad.png" /></span>' );
               }
--- ./patchsource/www/rega/esp/functions.fn.orig	2019-02-18 11:16:35.000000000 +0100
+++ ./patchsource/www/rega/esp/functions.fn	2018-12-01 10:11:26.000000000 +0100
@@ -331,6 +331,13 @@
     ! Hide channel 3 of this device
     bShowChannel = false;
   }
+  
+  ! HB Status Display EPaper 42BW
+  if(( (oDevHssType == "HB-DIS-EP-42BW") ) && (chNumber == 11)) {
+    ! Hide channel 11 of this device
+    bShowChannel = false;
+  }
+  
 }
 
 function ::showHideHmIPChannel() {
--- ./patchsource/www/webui/webui.js.orig	2019-02-18 11:16:40.000000000 +0100
+++ ./patchsource/www/webui/webui.js	2019-02-13 10:49:43.000000000 +0100
@@ -13146,7 +13146,7 @@
   // Check each line if the free text mode is active.
   // If yes switch on the appropriate text input field
   // and set the variable freeTextActive to true
-  for (var line = 0; line < 6; line++) {
+  for (var line = 0; line < 10; line++) {
     var jLine = jQuery("#textSelect_" + line),
     jCellFreeText = jQuery("#cellFreeText_"+line);
 
@@ -13162,7 +13162,7 @@
   // If at least one line has an active free text mode
   // we have to activate the placeholder for the other lines and the header (color and icon are moving up)
   if (freeTextActive) {
-    for (var line = 0; line < 6; line++) {
+    for (var line = 0; line < 10; line++) {
       var jLine = jQuery("#textSelect_" + line),
       jPlaceHolder = jQuery("#placeHolder_"+line);
 
@@ -13174,7 +13174,7 @@
     }
   } else {
     jHeadFreeText.hide().css("display", "none");
-    for (var line = 0; line < 6; line++) {
+    for (var line = 0; line < 10; line++) {
       jQuery("#placeHolder_"+line).hide().css("display", "none");
     }
   }
@@ -22539,6 +22539,12 @@
         this.setWidth(400);
         this.setFile( "/pages/msg/setOUCFMMode.htm" );
         break;
+        case ID_SET_HBOUCFM_MODE:
+        this.setTitle( translateKey("dialogEnterValue") /*"Wert eingeben" */);
+        this.addToPostBody( 'string type = "'+this.type+'";' );
+        this.setWidth(400);
+        this.setFile( "/pages/msg/setHBOUCFMMode.htm" );
+        break;
         case ID_SET_STATUS_DISPLAY:
         this.setTitle( translateKey("dialogEnterValue") /*"Wert eingeben" */);
         this.addToPostBody( 'string type = "'+this.type+'";' );
@@ -23337,6 +23343,7 @@
 ID_EDIT_SYSVAR           = 1021;
 ID_CHOOSE_LED            = 1022;
 ID_SET_OUCFM_MODE        = 1023;
+ID_SET_HBOUCFM_MODE      = 1923;
 ID_SET_STATUS_DISPLAY    = 1024;
 ID_BIDCOS_INTERFACE = 1025;
 
--- ./patchsource/www/rega/pages/tabs/admin/views/programs.htm.orig	2018-12-05 21:12:41.000000000 +0100
+++ ./patchsource/www/rega/pages/tabs/admin/views/programs.htm	2019-02-13 10:48:14.000000000 +0100
@@ -95,6 +95,19 @@
     };
   };  
 
+  // MP3 HB Funkgong
+  ChangeHBOUCFMValue = function(sdid)
+  {
+    iseSingleDestination.reload = false;
+    dlgResult = "";
+    CreatePopup(ID_SET_HBOUCFM_MODE, "X:"+sdid);
+    PopupClose = function()
+    {
+      dlgPopup.hide();
+      ReloadSingleDestination(sdid);
+    };
+  }; 
+  
   // Heating thermostate
   SetPartyMode = function(elm) {
     var url = "/pages/msg/setPartyMode.htm";
@@ -141,7 +154,10 @@
   };
 
   SetStatusDisplay = function(regaDialogID, mode) {
-    var elm = (mode == "DIS") ? jQuery("#setStatusDisplay" + regaDialogID) : jQuery("#setStatusDisplayEPaper" + regaDialogID) ;
+   var elm;
+   if (mode == "DIS") elm = jQuery("#setStatusDisplay" + regaDialogID);
+   if (mode == "DIS-EP") elm = jQuery("#setStatusDisplayEPaper" + regaDialogID) ;
+   if (mode == "DIS-EP42BW") elm = jQuery("#setHbStatusDisplayEPaper" + regaDialogID) ;
 
     jQuery(elm).prop("disabled", true);
 
@@ -179,6 +195,19 @@
             inputElm.change();
           }
         }, "html");
+      } else if (mode == "DIS-EP42BW") {
+         statusDisplayDialog = new HbStatusDisplayDialogEPaper(translateKey("statusDisplayDialogTitle"), htmlContent, jQuery(inputElm).val(), function (result) {
+          jQuery(elm).prop("disabled", false);
+          if (result == 1) {
+            conInfo("SetStatusDisplay: Yes pressed");
+            var inputElm = jQuery(elm).prev();
+            var cfgString = this.getConfigString();
+            conInfo(cfgString);
+            inputElm.val(cfgString); //.change();
+            encodeStringStatusDisplay(inputElm.prop("id"));
+            inputElm.change();
+          }
+        }, "html");
       }
 
 
--- ./patchsource/www/config/ic_common.tcl.orig	2018-12-05 21:12:41.000000000 +0100
+++ ./patchsource/www/config/ic_common.tcl	2018-12-02 10:57:56.000000000 +0100
@@ -1074,6 +1074,7 @@
 
   set hmDisEPIdentifier "HM-Dis-EP-WM55"
   set hmDisWM55Identifier "HM-Dis-WM55"
+  set hbDis42BWIdentifier "HB-DIS-EP-42BW"
 
   if { ! [catch { array set ch_descr [xmlrpc $iface_url($iface) getDeviceDescription [list string $address]] } ] } then {
     set channel_type $ch_descr(TYPE)
@@ -1167,41 +1168,48 @@
           append s "<td><input type=\"text\" name=\"$param_id\" value=\"$value\" $id $access /></td>"
         } else {
           puts "<script type=\"text/javascript\">load_JSFunc('/config/easymodes/MASTER_LANG/KEY_4Dis.js');</script>"
-          set helpText [getStatusDisplayHelp]
-
-          # The parameter numbering of the channels 1 and 2 are the same as from the HM-Dis-WM55
-          if {($parent_type != $hmDisEPIdentifier) || ($chn < 4)} {
+          set helpText [getStatusDisplayHelp]          
+          if {($parent_type == $hbDis42BWIdentifier)} then {
             if {$param_id == "TEXTLINE_1"} {
-              # Fortlaufende Nummerierung der Textblöcke hinzufügen.
-              # Berechnung:
-              # 1. Parameter = Kanalnummer * 2 - 1
-              if {($parent_type != $hmDisEPIdentifier) && ($parent_type != $hmDisWM55Identifier)} {
-                append s "<td>[expr $chn * 2 - 1]&nbsp;&nbsp;<input type=\"text\" name=\"$param_id\" maxlength=\"12\" onblur=\"encodeStringStatusDisplay('$idval', true);\" value=\"$value\" $id $access /></td>"
+              append s "<td>[expr $chn * 2 - 1]&nbsp;&nbsp;<input type=\"text\" name=\"$param_id\" maxlength=\"16\" onblur=\"encodeStringStatusDisplay('$idval', true);\" value=\"$value\" $id $access /></td>"
+            } else {
+              append s "<td>[expr $chn * 2]&nbsp;&nbsp;<input type=\"text\" name=\"$param_id\" maxlength=\"16\" onblur=\"encodeStringStatusDisplay('$idval', true);\" value=\"$value\" $id $access /></td>"
+            }
+          } else {
+            # The parameter numbering of the channels 1 and 2 are the same as from the HM-Dis-WM55
+            if {($parent_type != $hmDisEPIdentifier) || ($chn < 4)} {
+              if {$param_id == "TEXTLINE_1"} {
+                # Fortlaufende Nummerierung der Textblöcke hinzufügen.
+                # Berechnung:
+                # 1. Parameter = Kanalnummer * 2 - 1
+                if {($parent_type != $hmDisEPIdentifier) && ($parent_type != $hmDisWM55Identifier)} {
+                  append s "<td>[expr $chn * 2 - 1]&nbsp;&nbsp;<input type=\"text\" name=\"$param_id\" maxlength=\"12\" onblur=\"encodeStringStatusDisplay('$idval', true);\" value=\"$value\" $id $access /></td>"
+                } else {
+                  append s "<td>[expr $chn * 2 - 1]&nbsp;&nbsp;<input type=\"text\" name=\"$param_id\" maxlength=\"12\" onblur=\"encodeStringStatusDisplay('$idval', true, '_');\" value=\"$value\" $id $access /></td>"
+                }
               } else {
-                append s "<td>[expr $chn * 2 - 1]&nbsp;&nbsp;<input type=\"text\" name=\"$param_id\" maxlength=\"12\" onblur=\"encodeStringStatusDisplay('$idval', true, '_');\" value=\"$value\" $id $access /></td>"
+                # 2. Parameter = Kanalnummer * 2
+                if {($parent_type != $hmDisEPIdentifier) && ($parent_type != $hmDisWM55Identifier)} {
+                  append s "<td>[expr $chn * 2]&nbsp;&nbsp;<input type=\"text\" name=\"$param_id\" maxlength=\"12\" onblur=\"encodeStringStatusDisplay('$idval', true);\" value=\"$value\" $id $access /></td>"
+                } else {
+                  append s "<td>[expr $chn * 2]&nbsp;&nbsp;<input type=\"text\" name=\"$param_id\" maxlength=\"12\" onblur=\"encodeStringStatusDisplay('$idval', true, '_');\" value=\"$value\" $id $access /></td>"
+                }
+
+                append s "<td><img src=\"/ise/img/help.png\"/ size=\"24\" width=\"24\" onclick=\"MessageBox.show(translateKey('dialogHelpTitle'), '$helpText', '', 450, 375) ;\"></td>"
               }
             } else {
-              # 2. Parameter = Kanalnummer * 2
-              if {($parent_type != $hmDisEPIdentifier) && ($parent_type != $hmDisWM55Identifier)} {
-                append s "<td>[expr $chn * 2]&nbsp;&nbsp;<input type=\"text\" name=\"$param_id\" maxlength=\"12\" onblur=\"encodeStringStatusDisplay('$idval', true);\" value=\"$value\" $id $access /></td>"
+              # Here we set the parameter numbering for the channels 4 - 8 of the HM-Dis-EP-WM55
+              if {$param_id == "TEXTLINE_1"} {
+                # Fortlaufende Nummerierung der Textblöcke hinzufügen.
+                # Berechnung:
+                # 1. Parameter = Kanalnummer * 2 - 7
+                append s "<td>[expr $chn * 2 - 7]&nbsp;&nbsp;<input type=\"text\" name=\"$param_id\" maxlength=\"12\" onblur=\"encodeStringStatusDisplay('$idval', true, '_');\" value=\"$value\" $id $access /></td>"
               } else {
-                append s "<td>[expr $chn * 2]&nbsp;&nbsp;<input type=\"text\" name=\"$param_id\" maxlength=\"12\" onblur=\"encodeStringStatusDisplay('$idval', true, '_');\" value=\"$value\" $id $access /></td>"
+                # 2. Parameter = Kanalnummer * 2 -6
+                append s "<td>[expr $chn * 2 - 6]&nbsp;&nbsp;<input type=\"text\" name=\"$param_id\" maxlength=\"12\" onblur=\"encodeStringStatusDisplay('$idval', true, '_');\" value=\"$value\" $id $access /></td>"
+                append s "<td><img src=\"/ise/img/help.png\"/ size=\"24\" width=\"24\" onclick=\"MessageBox.show(translateKey('dialogHelpTitle'), '$helpText', '', 450, 375) ;\"></td>"
               }
-
-              append s "<td><img src=\"/ise/img/help.png\"/ size=\"24\" width=\"24\" onclick=\"MessageBox.show(translateKey('dialogHelpTitle'), '$helpText', '', 450, 375) ;\"></td>"
-            }
-          } else {
-            # Here we set the parameter numbering for the channels 4 - 8 of the HM-Dis-EP-WM55
-            if {$param_id == "TEXTLINE_1"} {
-              # Fortlaufende Nummerierung der Textblöcke hinzufügen.
-              # Berechnung:
-              # 1. Parameter = Kanalnummer * 2 - 7
-              append s "<td>[expr $chn * 2 - 7]&nbsp;&nbsp;<input type=\"text\" name=\"$param_id\" maxlength=\"12\" onblur=\"encodeStringStatusDisplay('$idval', true, '_');\" value=\"$value\" $id $access /></td>"
-            } else {
-              # 2. Parameter = Kanalnummer * 2 -6
-              append s "<td>[expr $chn * 2 - 6]&nbsp;&nbsp;<input type=\"text\" name=\"$param_id\" maxlength=\"12\" onblur=\"encodeStringStatusDisplay('$idval', true, '_');\" value=\"$value\" $id $access /></td>"
-              append s "<td><img src=\"/ise/img/help.png\"/ size=\"24\" width=\"24\" onclick=\"MessageBox.show(translateKey('dialogHelpTitle'), '$helpText', '', 450, 375) ;\"></td>"
-            }
+            }  
           }
         }
         append s "<td>$unit</td>"
--- ./patchsource/www/rega/esp/datapointconfigurator.fn.orig	2019-02-18 11:16:29.000000000 +0100
+++ ./patchsource/www/rega/esp/datapointconfigurator.fn	2019-01-17 13:28:45.000000000 +0100
@@ -99,6 +99,8 @@
     var CN_BRIGHTNESS_TRANSMITTER = "BRIGHTNESS_TRANSMITTER.";
     var CN_BTN_SHORT_ONLY  = "BTN_SHORT_ONLY.";
     var CN_BUTTON          = "BUTTON.";
+    var CN_RFID_BUTTON     = "RFIDBTN.";
+    var CN_IBTN_BUTTON     = "IBTN.";
     var CN_CAPACITIVE_FILLING_SENSOR  = "CAPACITIVE_FILLING_LEVEL_SENSOR.";
     var CN_CLIMATE_CONTROL_FLOOR_PUMP_TRANSCEIVER = "CLIMATECONTROL_FLOOR_PUMP_TRANSCEIVER.";
     var CN_CLIMATE_CONTROL_FLOOR_TRANSCEIVER = "CLIMATECONTROL_FLOOR_TRANSCEIVER.";
@@ -364,7 +366,27 @@
             sLastControlName = CN_BTN_SHORT_ONLY;
             Call("/esp/controls/button.fn::CreateKeyShortOnly()");
           }
+          
+          !# RFID
+          bIsControl = ( sControlName.Find(CN_RFID_BUTTON) > -1 );
+          if( bIsControl && (sLastControlName!=CN_RFID_BUTTON) )
+          {
+            WriteLine( "<script>conInfo('Control CN_RFID_BUTTON found.');</script>" );
+            isKnownControl = true;
+            sLastControlName = CN_RFID_BUTTON;
+            Call("/esp/controls/rfid_rc.fn::CreateRfidKey()");
+          }
 
+          !# iBUTTON
+          bIsControl = ( sControlName.Find(CN_IBTN_BUTTON) > -1 );
+          if( bIsControl && (sLastControlName!=CN_IBTN_BUTTON) )
+          {
+            WriteLine( "<script>conInfo('Control CN_IBTN_BUTTON found.');</script>" );
+            isKnownControl = true;
+            sLastControlName = CN_IBTN_BUTTON;
+            Call("/esp/controls/ibutton.fn::CreateIButtonKey()");
+          }
+          
           !# BUTTON
           bIsControl = ( sControlName.Find(CN_BUTTON) > -1 );
           if( bIsControl && (sLastControlName!=CN_BUTTON) )
--- ./patchsource/www/webui/js/lang/de/translate.lang.js.orig	2018-10-26 07:01:22.000000000 +0200
+++ ./patchsource/www/webui/js/lang/de/translate.lang.js	2019-01-16 13:44:23.000000000 +0100
@@ -764,6 +764,8 @@
     "actionStatusControlLblOff" : "Aus",
     "actionStatusControlLblShortKeyPress" : "Kurzer<br/>Tastendruck",
     "actionStatusControlLblLongKeyPress" : "Langer<br/>Tastendruck",
+    "actionStatusControlLblShortRfidHold" : "kurzer<br/>Scan",
+    "actionStatusControlLblLongRfidHold" : "langer<br/>Scan",
     "commentActionStatusControlEnd" : "***********************************************",
     "commentPrgRules" : "***********************************************",
     "programsCreateProgram" : "Programmerstellung",
--- ./patchsource/www/config/easymodes/SIGNAL_LEDM/signal_type.tcl.orig	2018-02-28 10:46:46.000000000 +0100
+++ ./patchsource/www/config/easymodes/SIGNAL_LEDM/signal_type.tcl	2019-02-11 15:29:09.000000000 +0100
@@ -1,7 +1,7 @@
 proc showExtendedColors {} {
   global dev_descr_receiver
   set result false
-  if {$dev_descr_receiver(PARENT_TYPE) == "HM-OU-CFM-TW"} {
+  if {$dev_descr_receiver(PARENT_TYPE) == "HM-OU-CFM-TW" || $dev_descr_receiver(PARENT_TYPE) == "HB-OU-MP3-LED"} {
     set result true
   }
   return $result
